haas/cli.py

```diff
@@ -135,25 +135,29 @@ def cmd(f):
     wrong number of arguments, and thirdly generates a 'usage' description and
     puts it in the usage dictionary.
     """
+
+    # Build the 'usage' info for the help:
+    args, varargs, _, _ = inspect.getargspec(f)
+    num_args = len(args)  # used later to validate passed args.
+    showee = [f.__name__] + ['<%s>' % name for name in args]
+    args = ' '.join(['<%s>' % name for name in args])
+    if varargs:
+        showee += ['<%s...>' % varargs]
+    usage_dict[f.__name__] = ' '.join(showee)
+
     @wraps(f)
     def wrapped(*args, **kwargs):
         try:
+            if len(args) < num_args or not varargs and len(args) > num_args:
```

Comment:
Can you please provide an explanation/comment for this, especially the ``not varargs`` part?